### DISCUSSION TITLE
Why flex-1 doesn't work?

### DISCUSSION THREAD

**Discussion by Rr7716** [tailwindlabs/tailwindcss#10600] [Help]
```js
<View className="flex-row pb-3 items-center mx-4 space-x-2">
          <Image
            source={{
              uri: "https://links.papareact.com/wru",
            }}
            className="h-7 w-7 bg-gray-300 p-4 rounded-full"
          />

          <View className="flex-1">
            <Text className="font-bold text-gray-400 text-xs">
              Deliver Now!
            </Text>
            <Text className="font-bold text-xl">
              Current Location
              <ChevronDownIcon size={20} color="#00CCBB" />
            </Text>
          </View>

          <UserIcon size={35} color="#00CCBB" />
</View>
```

Why UserIcon isn't on the right side? Can someone help me?

**Comment by wongjn** [comment]
`flex-1` applies the shorthand `flex` property but this only has an effect when the element with the `flex-1` is a child of a flex layout. Consider applying `display: flex` via the `flex` class to the parent of the `flex-1` element.

**Reply by Rr7716** [reply]
Thanks! It works when I apply it in the class of the parent!

**Reply by ege-cavus** [reply]
I've encountered the same problem, where did you apply the display: flex command exactly?

**Reply by wongjn** [reply]
To the direct parent of the element that has the `flex-1` class, like:

```html
<div class="flex">
  <div class="flex-1">
```

**Reply by ege-cavus** [reply]
Thank you for your help:)

**Reply by Abhi648** [reply]
hey I'm still facing this problem, can you guys please help me with the code or what needs to be changed
